src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -0,0 +1,306 @@
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(dataset_name="wikineural")
+
+        language_creators (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            with how the data was curated, such as `crowdsourced` or `machine_generated`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(language_creator="crowdsourced")
+
+        languages (:obj:`str` or :class:`List`, `optional`):
+            A string representing a two-character language to filter datasets by on the hub
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(language="en")
+
+        multilinguality (:obj:`str` or :class:`List`, `optional`):
+            A string representing a filter for datasets that contain multiple languages
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(multilinguality="yes")
+
+        size_categories (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the size of the dataset such as `100K<n<1M` or `1M<n<10M`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(size_categories="100K<n<1M")
+
+        task_categories (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the designed task, such as `audio_classification` or `named_entity_recognition`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(author="facebook")
+
+        task_ids (:obj:`str` or :class:`List`, `optional`):
+            A string or list of strings that can be used to identify datasets on the hub
+            by the specific task such as `speech_emotion_recognition` or `paraphrase`
+            Example usage:
+
+                >>> from huggingface_hub import DatasetFilter
+                >>> new_filter = DatasetFilter(task_ids="paraphrase")
+    """
+
+    author: str = None
+    benchmark: Union[str, List[str]] = None
+    dataset_name: str = None
+    language_creators: Union[str, List[str]] = None
+    languages: Union[str, List[str]] = None
+    multilinguality: Union[str, List[str]] = None
+    size_categories: Union[str, List[str]] = None
+    task_categories: Union[str, List[str]] = None
+    task_ids: Union[str, List[str]] = None
+
+
+@dataclass
+class ModelFilter:
+    """A class that converts human-readable model search parameters into ones compatible with
```

Comment:
Should this link to hf.co/models and mention the filters in the left?

Reply:
It should not in this case, as the filters available here are actually different than those that are provided in hf.co/models and hf.co/datasets. Considerably more with datasets in particular.